app/models/dataset_extra.rb

```diff
@@ -162,6 +162,7 @@ def decorate(dataset)
     decorate_modaps_layers(dataset)
 
     dataset[:links] = Array.wrap(dataset[:links]) # Ensure links attribute is present
+    dataset[:browseable_collection] = dataset[:id] unless dataset[:links].select{|link| link[:rel].include? 'browse'}.empty?
```

Comment:
This shouldn't be on DatasetExtra. It can be easily gleaned from the metadata on the client side.